src/Explain.h

```diff
@@ -305,19 +448,41 @@ class Explain {
         }
     }
 
-    // print string
-    void printStr(const std::string& s) {
-        if (ncurses && !output) {
-            wprintw(treePad, s.c_str());
-        } else {
-            if (output) {
-                *output << s;
-            } else {
-                std::cout << s;
-            }
+    /* Print any other information */
+    void printInfo(const std::string& info) override {
+        if (isatty(fileno(stdin))) {
+            wprintw(treePad, info.c_str());
+            prefresh(treePad, 0, 0, 0, 0, maxy - 3, maxx - 1);
         }
     }
 
+    /* Print an error, such as a wrong command */
+    void printError(const std::string& error) override {
+        wprintw(treePad, error.c_str());
+        prefresh(treePad, 0, 0, 0, 0, maxy - 3, maxx - 1);
+    }
+
+    /* Initialise ncurses window */
+    WINDOW* makeQueryWindow() {
+        WINDOW* w = newwin(3, maxx, maxy - 2, 0);
```

Comment:
`w` is not informative. You only need to remember what it means for 3 lines (at the moment), but you could also just call it something descriptive like `queryWindow`